src/lib.rs

```diff
@@ -46,6 +52,96 @@ pub fn mutate_request(mutated_object: serde_json::Value) -> wapc_guest::CallResu
     })?)
 }
 
+/// Update the pod sec from the resource defined in the original object
+/// and create an acceptance response.
+/// # Arguments
+/// * `validation_request` - the original admission request
+/// * `pod_spec` - new PodSpec to be set in the response
+pub fn mutate_pod_spec_from_request<T: std::default::Default>(
+    validation_request: ValidationRequest<T>,
+    pod_spec: PodSpec,
+) -> wapc_guest::CallResult {
+    match validation_request.request.kind.kind.as_str() {
+        Deployment::KIND => {
+            let mut deployment =
+                serde_json::from_value::<Deployment>(validation_request.request.object.clone())?;
+            if let Some(ref mut spec) = deployment.spec {
+                spec.template.spec = Some(pod_spec);
+            }
```

Comment:
It's a bit of a corner case, but if `deployment.spec` is `None`, then the mutation is not done.

I would replace the code to be something like:

``` rust
let mut deployment_spec = deployment.unwrap_or_default();
deployment_spec.template.spec = Some(pod_spec);
deployment.spec = Some(deployment_spec);
```

The same approach applies to the other types of resources being handled

Reply:
Done